processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -109,6 +118,27 @@ public SearchQueryRunner(Segment segment)
       final ImmutableBitmap baseFilter =
           filter == null ? null : filter.getBitmapIndex(new ColumnSelectorBitmapIndexSelector(bitmapFactory, index));
 
+      ImmutableBitmap timeFilteredBitmap;
+      if (!interval.contains(segment.getDataInterval()))
+      {
+        MutableBitmap timeBitmap = bitmapFactory.makeEmptyMutableBitmap();
+        final Column timeColumn = index.getColumn(Column.TIME_COLUMN_NAME);
+        final GenericColumn timeValues = timeColumn.getGenericColumn();
+
+        int startIndex = Math.max(0, getStartIndexOfTime(timeValues, interval.getStartMillis(), true));
+        int endIndex = Math.min(timeValues.length() - 1, getStartIndexOfTime(timeValues, interval.getEndMillis(), false));
+
+        for (int i = startIndex; i <= endIndex; i++) {
+          timeBitmap.add(i);
+        }
+
+        final ImmutableBitmap finalTimeBitmap = bitmapFactory.makeImmutableBitmap(timeBitmap);
+        timeFilteredBitmap =
+            (baseFilter == null) ? finalTimeBitmap : finalTimeBitmap.intersection(baseFilter);
```

Comment:
@fjy Thank you for the comment and sorry for too late response.
I think building a bitmap index is inevitable to fix bug and hard to imagine the target approach to compare the performance.
Would you advice me to what I should compare the performance of my approach?

By the way, it is bad that MutableBitmap interface does not provide range update method like flip() or fill() even though all its underlying bitmap implementations (RoaringBitmap, BitSet and ConciseSet) support at least one of them.